site/content/en/docs/technical-outcomes/scalability/scale-using-vpa/scale-using-vpa.demo

```diff
@@ -0,0 +1,38 @@
+# Scale pod CPU request up & down using VPA
+
+# Get metrics usage
+wget https://github.com/kubernetes-sigs/kwok/releases/download/v0.6.0/metrics-usage.yaml
+
+# Set up Cluster
+kwokctl create cluster --enable-metrics-server --config ${WORK_DIR}/kwokctl.yaml --config ./metrics-usage.yaml --runtime binary
+
+# Create a node
+kwokctl scale node --replicas 1 --param '.allocatable.cpu="4000m"'
+
+# Set up VPA
+git clone https://github.com/kubernetes/autoscaler.git
+kubectl apply -f ./autoscaler/vertical-pod-autoscaler/deploy/vpa-v1-crd-gen.yaml
+kubectl apply -f ./autoscaler/vertical-pod-autoscaler/deploy/vpa-rbac.yaml
+
+# Deploy the three components of the Vertical Pod Autoscaler
+{ cd autoscaler/vertical-pod-autoscaler/ && NAMESPACE=kube-system go run ./pkg/admission-controller --kubeconfig ~/.kube/config --client-ca-file ~/.kwok/clusters/kwok/pki/ca.crt --tls-cert-file  ~/.kwok/clusters/kwok/pki/admin.crt --tls-private-key  ~/.kwok/clusters/kwok/pki/admin.key --webhook-address https://127.0.0.1 --webhook-port 8080 --register-by-url --port 8080 ;} &
+{ cd autoscaler/vertical-pod-autoscaler/ && NAMESPACE=kube-system go run ./pkg/recommender --kubeconfig ~/.kube/config ;} &
+{ cd autoscaler/vertical-pod-autoscaler/ && NAMESPACE=kube-system go run ./pkg/updater --kubeconfig ~/.kube/config ;} &
+
+# Create the VPA
+kubectl apply -f ${WORK_DIR}/vpa.yaml
+
+# Create deployment
+kubectl apply -f ${WORK_DIR}/deployment.yaml
+
+# Wait for 180s
+sleep 180
```

Comment:
The new command keeps timing out because its default value is `30s` but the VPA recommendation condition takes longer than that.
The error message encountered is:
```bash
error: timed out waiting for the condition on verticalpodautoscalers/fake-vpa
```

With the old command, a user could indeed miss it. When I first tried the simulation, `180s` was the estimated time for the VPA to provide a recommendation. But today, it took about `80s`. So I added a `--timeout` flag.
```bash
kubectl wait --for=condition=RecommendationProvided=true vpa/fake-vpa --timeout=80s
```
This adjustment worked after a few attempts, but the command still occasionally times out. Based on current observations, `180s` is the ideal duration needed for the VPA to provide a recommendation.

Reply:
Here is my suggestion 
```bash

# Wait until the recommendation is provided
kubectl wait --for=condition=RecommendationProvided=true vpa/fake-vpa --timeout=120s

# Check CPU usage
kubectl top pod

# Check recommended VPA values
kubectl describe vpa fake-vpa | awk '/Recommendation:/,/Events:/' | sed '$d'

# Wait 60s
sleep 60

## See the new applied limits
kubectl get pod -o yaml | grep -i cpu:

```